File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.js
function solution(video_len, pos, op_start, op_end, commands) {
    // video_len
    const splitedVideoLen = video_len.split(":");
    const videoLen = parseInt(splitedVideoLen[0]) * 60 + parseInt(splitedVideoLen[1]);
    // pos
    const splitedPosLen = pos.split(":");
    let posLen = parseInt(splitedPosLen[0]) * 60 + parseInt(splitedPosLen[1]);
    // op_start
    const splitedOpStartLen = op_start.split(":");
    const opStartLen = parseInt(splitedOpStartLen[0]) * 60 + parseInt(splitedOpStartLen[1]);
    // op_end
    const splitedOpEndLen = op_end.split(":");
    const opEndLen = parseInt(splitedOpEndLen[0]) * 60 + parseInt(splitedOpEndLen[1]);
    
    for (const command of commands) {
        if (posLen >= opStartLen && posLen <= opEndLen) posLen = opEndLen;
        
        if (command === "next") {
            const nextPosLen = posLen + 10;
            
            if (nextPosLen >= videoLen) {
                posLen = videoLen;
            } else {
                posLen = nextPosLen;
            }
        } else {
            const prevPosLen = posLen - 10;
            
            if (prevPosLen <= 0) {
                posLen = 0;
            } else {
                posLen = prevPosLen;
            }
        }
        
        if (posLen >= opStartLen && posLen <= opEndLen) posLen = opEndLen;
    }
    
    let answer = "";
    const hour = Math.floor(posLen / 60).toString().length === 1 ? "0" + Math.floor(posLen / 60).toString() : Math.floor(posLen / 60).toString();
    const minute = (posLen % 60).toString().length === 1 ? "0" + (posLen % 60).toString() : (posLen % 60).toString();
    answer = hour + ":" + minute;
    
    return answer;
}

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
def change_time(time):
    splited_time = time.split(":")
    minute, second = int(splited_time[0]), int(splited_time[1])
    return minute * 60 + second

def return_change_time(time):
    minute = str(time // 60)
    second = str(time % 60)
    return f"{minute.zfill(2)}:{second.zfill(2)}"
    

def solution(video_len, pos, op_start, op_end, commands):
    # change time minute to second
    changed_video_len, changed_pos, changed_op_start, changed_op_end = change_time(video_len), change_time(pos), change_time(op_start), change_time(op_end)
    
    # start commands
    for command in commands:
        # 오프닝 건너뛰기
        if changed_pos >= changed_op_start and changed_pos <= changed_op_end:
            changed_pos = changed_op_end
        
        # prev
        if command == "prev":
            if changed_pos < 10:
                changed_pos = 0
            else:
                changed_pos -= 10
                
        # next
        if command == "next":
            if changed_pos + 10 >= changed_video_len:
                changed_pos = changed_video_len
            else:
                changed_pos += 10
                
        # 오프닝 건너뛰기
        if changed_pos >= changed_op_start and changed_pos <= changed_op_end:
            changed_pos = changed_op_end
            
    return return_change_time(changed_pos)